config/cylc_hpc/flow.cylc

```diff
@@ -42,40 +49,99 @@
+        execution time limit = PT1H
+        [[[directives]]]
+            --mem = 12G
+            --cpus-per-task = 8
+    
     [[pulljuliaimage]]
         script = """
-            apptainer build --force $project_dir/icefloetracker-julia.simg docker://brownccv/icefloetracker-julia:pr-45
+            apptainer pull --force $project_dir/icefloetracker-julia.simg docker://brownccv/icefloetracker-julia:pr-45
         """
+        platform = oscar
+        execution time limit = PT1H
+        [[[directives]]]
+            --mem = 16G
+            --cpus-per-task = 8
+    
     [[landmask]]
         script = """
             singularity exec --bind $landmask_dir:/tmp,$report_dir:/usr/local/bin/../report $project_dir/icefloetracker-julia.simg $julia_exec -t auto /usr/local/bin/ice-floe-tracker.jl landmask $fetchdata_dir /tmp
         """
+        platform = oscar
+        execution time limit = PT1H
+        [[[directives]]]
+            --mem = 20G
+            --cpus-per-task = 16
+    
     [[preprocess]]
         script = """
             singularity exec --bind $preprocess_dir:/tmp,$report_dir:/usr/local/bin/../report $project_dir/icefloetracker-julia.simg $julia_exec -t auto /usr/local/bin/ice-floe-tracker.jl preprocess -t $fetchdata_dir/truecolor -r $fetchdata_dir/reflectance -l $landmask_dir -p $soit_dir -o /tmp
         """
+        platform = oscar
+        execution time limit = PT1H
+        [[[directives]]]
+            --mem = 36G
+            --cpus-per-task = 40
+    
     [[extractfeatures]]
         script = """
             singularity exec --bind $preprocess_dir:/tmp,$report_dir:/usr/local/bin/../report $project_dir/icefloetracker-julia.simg $julia_exec -t auto /usr/local/bin/ice-floe-tracker.jl extractfeatures -i $preprocess_dir -o /tmp --minarea $minfloearea --maxarea $maxfloearea
         """
+        platform = oscar
+        execution time limit = PT1H
+        [[[directives]]]
+            --mem = 20G
+            --cpus-per-task = 8
+    
     [[tracking]]
         script = """
             singularity exec --bind $tracker_dir:/tmp,$report_dir:/usr/local/bin/../report $project_dir/icefloetracker-julia.simg $julia_exec -t auto /usr/local/bin/ice-floe-tracker.jl track --imgs $preprocess_dir --props $preprocess_dir --deltat $preprocess_dir --output /tmp
         """
+        platform = oscar
+        execution time limit = PT1H
+        [[[directives]]]
+            --mem = 20G
+            --cpus-per-task = 8
+    
     [[exportH5]]
         script = """
             singularity exec --bind $preprocess_dir:/tmp,$report_dir:/usr/local/bin/../report $project_dir/icefloetracker-julia.simg $julia_exec -t auto /usr/local/bin/ice-floe-tracker.jl makeh5files --pathtosampleimg $fetchdata_dir/truecolor/$(ls $fetchdata_dir/truecolor | head -1) --resdir /tmp
         """
+        platform = oscar
+        execution time limit = PT1H
+        [[[directives]]]
+            --mem = 32G
+            --cpus-per-task = 24
```

Comment:
So this kind of thing cannot be set globally?

Reply:
I will add some defaults to the global.cylc file and then we can also specify directives in each block to override the defaults.

Reply:
good point, but it's such a tiny task that maybe it's okay?